### PR TITLE
Fix onClick behavior in Menu

### PR DESCRIPTION
**Pull request by taysea**
#### What does this PR do?
This PR fixes the cross browser onClick functionality that was removed in the previous update to Menu. The mouse can be used to open and select menu items. For keyboard accessibility, space/enter/down/up can be used to open the menu and tab can be used to navigate into and out of the menu. If the user tabs out of the menu, the menu will close. These functions work on Chrome, Safari, FireFox, Edge, and IE11.

#### Where should the reviewer start?
src/js/components/Menu/Menu.js

#### What testing has been done on this PR?
These changes have been tested by creating an updated test using react testing library. They have also been tested manually by adding a console statement when the onClick handler of a menu item is called. This console statement printed out which item was clicked to ensure the correct item was being registered. This was tested using both mouse and keyboard.

#### How should this be manually tested?
To manually test, you can add this line `console.log(item)` to line 232 of Menu.js and then interact with the Menu as noted above.

#### Any background context you want to provide?
In the process of trying to improve keyboard accessibility on the Menu component, the onClick functionality was affected. This PR fixes the regression while also maintaining some of the improved accessibility. Future steps will need to be taken to improve accessibility for arrow keys.

#### What are the relevant issues?
Issue #3408 

#### Do the grommet docs need to be updated?
No.

#### Should this PR be mentioned in the release notes?
No.

#### Is this change backwards compatible or is it a breaking change?
Yes, backwards compatible.

### CODE REVIEW

**Comment by ShimiSun** [comment]
linking to #3334

File: src/js/components/Menu/__tests__/Menu-test.js [hunk 1]
```diff
@@ -140,39 +140,36 @@ describe('Menu', () => {
...
-
-    // moves to the first suggestion
-    fireEvent.keyDown(getByLabelText('Open Menu'), {
-      key: 'Up',
-      keyCode: 38,
-      which: 38,
+    fireEvent.keyDown(getByLabelText('Close Menu'), {
+      key: 'Tab',
+      keyCode: 9,
+      which: 9,
```
Comment (ShimiSun) [inline src/js/components/Menu/__tests__/Menu-test.js]: Are we closing the Menu three times using Tab? 
Before these changes, it made sense to use the Tab in order to navigate within the items of the 'Open Menu', but when the Menu is closed how does that makes sense?

File: src/js/components/Menu/__tests__/__snapshots__/Menu-test.js.snap [hunk 2]
```diff
@@ -2492,9 +2489,6 @@ exports[`Menu open and close on click 3`] = `
   border: none;
   padding: 0;
   text-align: inherit;
-  background-color: rgba(221,221,221, 0.4);
```
Comment (ShimiSun) [inline src/js/components/Menu/__tests__/__snapshots__/Menu-test.js.snap]: Please, doubles check all these styles didn't exist prior to your previous changes as well. I want to make sure we are not introducing regression from the style aspect.

File: src/js/components/Menu/__tests__/Menu-test.js [hunk 3]
```diff
@@ -140,39 +140,36 @@ describe('Menu', () => {
...
     );
     expect(container.firstChild).toMatchSnapshot();
 
-    // pressing down 3x: first opens the drop,
-    // second moves to the first suggestion
-    // third moves to the last suggestion
+    // pressing space: opens drop
+    // tab moves to menu label
+    // second moves to first item
+    // third moves to second item
```
Comment (ShimiSun) [inline src/js/components/Menu/__tests__/Menu-test.js]: This description no longer describes the current code flow, I think it should be adjusted.

File: src/js/components/Menu/Menu.js [hunk 4]
```diff
@@ -45,6 +45,10 @@ To make a selection:
 - Space is pressed.
 */
 
+const INDEX = {
+  NONE: 'none',
```
[the remark this answers is not in the document]
Comment (taysea) [reply src/js/components/Menu/Menu.js]: You're right, we don't the the extra abstraction. I will change it.

File: src/js/components/Menu/Menu.js [hunk 5]
```diff
@@ -71,54 +75,45 @@ const Menu = props => {
...
-          buttonRefs[items.length].focus();
-          setActiveItemIndex(-1);
-        }
-      }
-    }
-  }, [isManagingFocus]);
 
   const onDropClose = () => {
-    setActiveItemIndex(-1);
+    setActiveItemIndex(INDEX.NONE);
```
[the remark this answers is not in the document]
Comment (taysea) [reply src/js/components/Menu/Menu.js]: When the menu opens with the mouse, I don't believe we want any of the menu buttons to have an active state styling. If the active item index is -1 by default, when the menu has `align.top === 'top'`, then the menu label button will receive active styling upon opening the menu even if it not being hovered over. By making the default active index NONE, this styling will not occur until keyboard is used to interact with the menu or the mouse hovers over a menu item.

File: src/js/components/Menu/__tests__/Menu-test.js (hunk 1 above)
Comment (taysea) [reply src/js/components/Menu/__tests__/Menu-test.js]: We're not closing the menu, but the 'Close Menu' button refers to the Drop div that gets added to the DOM when a menu opens, and it seemed that the keyDown focus needed to refer to this element in order to advance through the menu items which are contained in this Drop. So, the menu is not closing but using that label text is helping to tell the test where the keyDown should occur. I am still strengthening my understanding of test writing, so another set of eyes may be helpful.

File: src/js/components/Menu/__tests__/Menu-test.js (hunk 3 above)
Comment (taysea) [reply src/js/components/Menu/__tests__/Menu-test.js]: This description does refer to the expected flow, but if this is not what is happening please let me know. But this describes the test steps as:
1. space opens the menu
2. tab to the menu label
3. tab to the first item
4. tab to the second item
Then, the second item is selected with 'Enter'.

File: src/js/components/Menu/__tests__/__snapshots__/Menu-test.js.snap (hunk 2 above)
Comment (taysea) [reply src/js/components/Menu/__tests__/__snapshots__/Menu-test.js.snap]: I confirmed these styles didn't exist prior to my previous changes.

**Comment by ShimiSun** [comment]
Hi there! Not sure why all my comments weren't posted yesterday, sorry for the late post

**Comment by ShimiSun** [comment]
Actually, I think we are working on double threads with a similar PR #3464. Can we close this PR?

**Pull request closed without merging**